### ISSUE TITLE
File upload size limit for each user/group

### ISSUE THREAD

**Issue by pandurx** [Pandurx-Elgg/c_dashboard_usage#4]

**Comment by bartjkdp** [comment]
I can manage this. I think it is difficult to enforce a hard limit as we have to overwrite file uploading logic. This could be implemented (depending on the plugins of the site) both in file and file_tools. I propose a soft limit with warning messages. My plan is to add a hook to a file create or update event and check the quota of the user. How and where is the quota configured?

**Comment by bartjkdp** [comment]
Wait a minute, maybe hooking on the file update and create event could still be possible to enforce a hard limit. Will do some experimentation tomorrow.

**Comment by pandurx** [comment]
Can't we do a check of the file limit before letting user upload file?

**Comment by bartjkdp** [comment]
Now I know what you mean. The max file size upload is already handled by PHP/Elgg. You can configure it in .htaccess

```
    php_value post_max_size 15M
    php_value upload_max_filesize 10M
```

I was thinking of a specific quota per user and per group for the total amount of storage a group or user can have.

**Comment by pandurx** [comment]
We can do a soft limit. 
Maybe a user/group metadata to store the limit and then check that first before uploading?

**Comment by bartjkdp** [comment]
That's a good idea! Will work on that.

**Comment by bartjkdp** [comment]
Hard quota limit is just implemented in 13ded2c
